luxonis_ml/nn_archive/config_building_blocks/base_models/head.py

```diff
@@ -190,93 +148,146 @@ def validate_label_type(
+    )
+    is_softmax: Optional[bool] = Field(
+        None,
+        description="True, if output is already softmaxed in YOLO instance segmentation.",
     )
-    n_keypoints: int = Field(description="Number of keypoints per bbox.")
 
     @field_validator("family")
     def validate_label_type(
         cls,
         value,
     ):
-        if value != "KeypointDetectionYOLO":
+        if value not in [
+            "ObjectDetectionYOLO",
+            "InstanceSegmentationYOLO",
+            "KeypointDetectionYOLO",
+            "OBBDetectionYOLO",
+        ]:
             raise ValueError("Invalid family")
         return value
 
+    @model_validator(mode="before")
+    def validate_task_specific_fields(
+        cls,
+        values,
+    ):
+        family = values.get("family")
+
+        required_fields = {
+            "InstanceSegmentationYOLO": [
+                "postprocessor_path",
+                "n_prototypes",
+                "is_softmax",
+            ],
+            "KeypointDetectionYOLO": ["n_keypoints"],
+        }
+
+        unsupported_fields = {
+            "ObjectDetectionYOLO": [
+                "postprocessor_path",
+                "n_prototypes",
+                "n_keypoints",
+                "is_softmax",
+            ],
+            "OBBDetectionYOLO": [
+                "postprocessor_path",
+                "n_prototypes",
+                "n_keypoints",
+                "is_softmax",
+            ],
+            "InstanceSegmentationYOLO": ["n_keypoints"],
+            "KeypointDetectionYOLO": [
+                "postprocessor_path",
+                "n_prototypes",
+                "is_softmax",
+            ],
+        }
+
+        for field in required_fields.get(family, []):
+            if field not in values or values[field] is None:
+                raise ValueError(f"{family} requires {field}.")
+
+        for field in unsupported_fields.get(family, []):
+            if field in values and values[field] is not None:
+                raise ValueError(f"{family} does not support {field}.")
 
-class HeadOBBDetectionYOLO(HeadObjectDetectionYOLO, ABC):
-    """Metadata for YOLO OBB head.
-
-    @type family: str
-    @ivar family: Decoding family.
-    @type outputs: C{OutputsOBBDetectionYOLO}
-    @ivar outputs: A configuration specifying which output names from the `outputs` block of the archive are fed into the head.
-    """
+        return values
 
-    family: Literal["OBBDetectionYOLO"] = Field(..., description="Decoding family.")
-    outputs: OutputsOBBDetectionYOLO = Field(
-        description="A configuration specifying which output names from the `outputs` block of the archive are fed into the head."
-    )
+    @model_validator(mode="before")
+    def validate_anchors(cls, values):
+        if (
+            "anchors" in values
+            and values["anchors"] is not None
+            and values["subtype"] == ObjectDetectionSubtypeYOLO.YOLOv6
```

Comment:
I believe v8 as well?

Reply:
Added in [e6ddc8e](https://github.com/luxonis/luxonis-ml/pull/112/commits/e6ddc8ec4d9199bb47d82535b71fe6fba3c3b6cf).